crates/accelerate/src/commutation_cancellation.rs

```diff
@@ -0,0 +1,285 @@
+                        Some(Param::Float(f)) => *f,
+                        _ => panic!(
+                            "Rotational gate with parameter expression encoutned in cancellation"
+                        ),
+                    }
+                } else if HALF_TURNS.contains(&node_op_name) {
+                    PI
+                } else if QUARTER_TURNS.contains(&node_op_name) {
+                    PI / 2.0
+                } else if EIGHTH_TURNS.contains(&node_op_name) {
+                    PI / 4.0
+                } else {
+                    panic!("Angle for operation {node_op_name} is not defined")
+                };
+                total_angle += node_angle;
+
+                if let Some(definition) = node_op.op.definition(node_op.params_view()) {
+                    //TODO check for PyNone global phase?
+                    //total_phase += match definition.global_phase {Param::Float(f) => f, Param::Obj(pyop) => , Param::ParameterExpression(_) => panic!("PackedInstruction with definition has global phase set as parameter expression")};
+                    total_phase += match definition.global_phase {Param::Float(f) => f, _ => panic!("PackedInstruction with definition has no global phase set as floating point number")};
+                }
+            }
+
+            let new_op = if cancel_key.0 == Z_ROTATION {
+                z_var_gate.unwrap()
+            } else if cancel_key.0 == X_ROTATION {
+                &RXGate
+            } else {
+                panic!("impossible case!");
+            };
+
+            let gate_angle = mod_2pi(total_angle, 0.);
+
+            let new_op_phase = if gate_angle.abs() > _CUTOFF_PRECISION {
+                let new_index = dag.insert_1q_on_incoming_qubit(
+                    (*new_op, &[total_angle]),
+                    *cancel_set.first().unwrap(),
+                );
+                let new_node = match &dag.dag[new_index] {
+                    NodeType::Operation(instr) => instr,
+                    _ => panic!("Unexpected type in commutation set run."),
+                };
+
+                if let Some(definition) = new_node.op.definition(new_node.params_view()) {
+                    //TODO check for PyNone global phase?
+                    match definition.global_phase {Param::Float(f) => f, _ => panic!("PackedInstruction with definition has no global phase set as floating point number")}
+                } else {
+                    0.0f64
+                }
+            } else {
+                0.0f64
+            };
+
+            dag.add_global_phase(py, &Param::Float(total_phase - new_op_phase))?;
+
+            for node in cancel_set {
+                dag.remove_op_node(*node);
+            }
+
+            //TODO do we need this due to numerical instability?
+            /*
+                if np.mod(total_angle, (2 * np.pi)) < _CUTOFF_PRECISION:
+                    dag.remove_op_node(run[0])
+            */
```

Comment:
In it's current form this isn't needed since you're removing all the gates regardless of the angle so I would just delete the comment. If this is needed we'd have to update the removal loop to account for this.

```suggestion
```

Reply:
Yeah, I did not understand why it would be needed in the original code and thought it may be related to numerical instability. I will remove it.